transactions/idTableStaking/admin/capability_end_epoch.cdc

```diff
@@ -0,0 +1,39 @@
+import FlowIDTableStaking from 0xIDENTITYTABLEADDRESS
+
+// This transaction uses a staking admin capability
+// to pay rewards, end the staking auction, and end the epoch.
+//
+// It combines the pay_rewards, end_staking and move_tokens transactions
+// which ends the staking auction, which refunds nodes with insufficient stake
+// and moves tokens between buckets
+// It also sets a new token payout for the next epoch
+
+transaction(ids: [String], newPayout: UFix64) {
+
+    // Local variable for a reference to the ID Table Admin object
+    let adminRef: &FlowIDTableStaking.Admin
+
+    prepare(acct: AuthAccount) {
+        let adminCapability = acct.copy<Capability>(from: FlowIDTableStaking.StakingAdminStoragePath)
+            ?? panic("Could not get capability from account storage")
+
+        // borrow a reference to the admin object
+        self.adminRef = adminCapability.borrow<&FlowIDTableStaking.Admin>()
+            ?? panic("Could not borrow reference to staking admin")
+    }
+
+    execute {
+        let approvedIDs: {String: Bool} = {}
+        for id in ids {
+            approvedIDs[id] = true
+        }
```

Comment:
Is there any reason why the dictionary isn't passed in as an argument, but constructed from an array?

Reply:
It is just because when I first wrote the transaction like 8 months ago, I didn't know how to construct a dictionary in Go. Should I update it?

Reply:
Ah, no worries! Yeah, I think it would be good to update it, I can help with hit

Reply:
BTW, no need to do it in this PR, can be done separately, this works